pkg/app/piped/controller/controller.go

```diff
@@ -643,6 +671,19 @@ func (c *controller) getMostRecentlySuccessfulDeployment(ctx context.Context, ap
 	return nil, err
 }
 
+func (c *controller) shouldStartPlanningDeployment(ctx context.Context, d *model.Deployment) (bool, error) {
+	if !d.IsInChainDeployment() {
+		return true, nil
+	}
+	resp, err := c.apiClient.InChainDeploymentPlannable(ctx, &pipedservice.InChainDeploymentPlannableRequest{
+		Deployment: d,
```

Comment:
Sorry I have just found a thing.
Do you need to send the whole deployment object?
It might be a bit redundant.

Reply:
To determine should we plan that deployment, currently we need: DeploymentChainId and DeploymentChainBlockIndex, not the whole deployment object. Should we only submit those 2 fields? I'm okay with both but in the future, if we need more than that to determine if this deployment should be planed or not, we have to update RPC too.

Reply:
I see.
Let's change the RPC to use just needed fields.

Reply:
sure 🙆‍♂️

Reply:
Addressed by https://github.com/pipe-cd/pipe/pull/2844/commits/87054ac3ef9c46cde2b0c19bb10a098f01b66602